extras/nginx_config_model.txt
-- Remove the history from
rm -rf .git

-- recreate the repos from the current content only
git init
git add .
git commit -m "Initial commit"

-- push to the github remote repos ensuring you overwrite history
git remote add origin git@github.com:<YOUR ACCOUNT>/<YOUR REPOS>.git
git push -u --force origin master

-------------------------------------------------------------
find . -path "*/migrations/*.py" -not -name "__init__.py" -delete
find . -path "*/migrations/*.pyc"  -delete



--------------------------------------------------
    publicacion = {
        "camara_serial": method.routing_key.split('.')[2],
        "zona": method.routing_key.split('.')[3],
        "topico=": method.routing_key,
        "mensaje": body.decode('ascii'),
        "nro_personas": nro_personas,
        "fecha_hora": d
        # "qos=": message.qos,
        # "retain_flag": message.retain
    }


==============================================
Django recommends that you deploy static files with a web server other than wsgi.

In settings.py, set:
STATIC_ROOT = 'static'

Run python manage.py collectstatic, which will copy the Django admin static files to /path/to/project/static/

Configure your static file server. If you use Nginx, you could add this config:

location /static/ {
    alias /path/to/project/static/;
    expires modified +1w;
}
Reload your web server

You should now have access to the static files.



======================================================
sudo fuser -k 8081/tcp

<form class="success" id="form-id" action="{% url "formly_dt_page_create" pk=selected_survey.pk %}" method="post">
    {% csrf_token %}
    <a class="nav-link {% if page == selected_page %}active{% endif %}" onclick="document.getElementById('form-id').submit();">Add page</a>
</form>

project's root directory with a command similar to this:
gunicorn -c gunicorn/gunicorn-config.py your-project.wsgi
gunicorn -c gunicorn/gunicorn-config.py core.wsgi

lsof | grep nohup.out

nohup gunicorn -c gunicorn/gunicorn-config.py core.wsgi &

exec gunicorn wsgi:app -u gunicorn -g nginx -m 007 -b gunicorn.sock >> /var/log/$<service_name>.sys.log 2>&1

exec gunicorn wsgi:app -u gunicorn -g nginx -m 007 -b gunicorn.sock >> /var/log/$<service_name>.sys.log 2>&1

===================================================

db.copyDatabase( "test", "records", "db0.example.net", "<username>", "<password>")

db.dropDatabase("SMF_BDControlAforo_V1")

db.copyDatabase( "SMF_BDControlAforo", "SMF_BDControlAforo", "5.196.27.225:27117", "smfactory", "S0ftwar4Mf4ac0ory")

mongodump --archive --db=SMF_BDControlAforo | mongorestore --archive  --nsFrom='test.*' --nsTo='examples.*'


db.cloneDatabase("5.196.27.225:27117")


===================================================

#user  nginx;
worker_processes  1;

error_log  /var/log/nginx/error.log;
error_log  /var/log/nginx/error.log  notice;
error_log  /var/log/nginx/error.log  info;

#pid        /var/run/nginx.pid;

include /etc/nginx/modules.conf.d/*.conf;

events {
    worker_connections  1024;
}


http {
    include       mime.types;
    default_type  application/octet-stream;

    log_format  main  '$remote_addr - $remote_user [$time_local] "$request" '
                      '$status $body_bytes_sent "$http_referer" '
                      '"$http_user_agent" "$http_x_forwarded_for"';

    access_log  /var/log/nginx/access.log  main;

    sendfile        on;
    #tcp_nopush     on;

    #keepalive_timeout  0;
    keepalive_timeout  65;
    #tcp_nodelay        on;

    #gzip  on;
    #gzip_disable "MSIE [1-6]\.(?!.*SV1)";

    server_tokens off;

    include /etc/nginx/conf.d/*.conf;
}

# override global parameters e.g. worker_rlimit_nofile
include /etc/nginx/*global_params;
It worked



=========================================================

upstream merakimvsense-gunicorn {
	server unix:/tmp/gunicorn.sock fail_timeout=0;
}

server {

	listen 80 default;
	client_max_body_size 4G;
	server_name your-site.com;
	keepalive_timeout 70;

	access_log /var/log/nginx/access_log;
	error_log /var/log/nginx/error_log;

	root /opt/merakimvsense/core;

	location /static/ {
		autoindex on;
		alias /opt/merakimvsense/core/static/;
		expires 1M;
		access_log off;
		add_header Cache-Control "public";
		proxy_ignore_headers "Set-Cookie";
	}

	location @proxy_merakimvsense {
		proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
		proxy_set_header Host $http_host;
		proxy_redirect off;
		proxy_pass merakimvsense-gunicorn;
	}

	location / {
		try_files $uri @proxy_merakimvsense;
	}
}


=======================================
working  etc/nginx/sites/availables
=======================================
upstream merakimvsense-gunicorn {
	server unix:/tmp/gunicorn.sock fail_timeout=0;
}

server {

	listen 80 default;
	client_max_body_size 4G;
	server_name your-site.com;
	keepalive_timeout 70;

	access_log /var/log/nginx/access_log;
	error_log /var/log/nginx/error_log;

	root /opt/merakimvsense/core;

	location /static/ {
		autoindex on;
		alias /opt/merakimvsense/core/static/;
		expires 1M;
		access_log off;
		add_header Cache-Control "public";
		proxy_ignore_headers "Set-Cookie";
	}

	location @proxy_merakimvsense {
		proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
		proxy_set_header Host $http_host;
		proxy_redirect off;
		proxy_pass http://merakimvsense-gunicorn;
	}

	location / {
		try_files $uri @proxy_merakimvsense;
	}
}


core/settings.py
# -*- encoding: utf-8 -*-
"""
License: MIT
Copyright (c) 2019 - present AppSeed.us
"""
from pymongo import MongoClient
import os
from decouple import config
from unipath import Path
import dj_database_url

# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
BASE_DIR    = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
PROJECT_DIR = Path(__file__).parent

# SECURITY WARNING: keep the secret key used in production secret!
SECRET_KEY = config('SECRET_KEY', default='S#perS3crEt_1122')

# SECURITY WARNING: don't run with debug turned on in production!
#DEBUG = config('DEBUG', default=False)
DEBUG = config('DEBUG', default=True)

# load production server from .env
ALLOWED_HOSTS = ['localhost', '127.0.0.1', '5.196.27.225', config('SERVER', default='127.0.0.1')]

CORS_ORIGIN_ALLOW_ALL = False
CORS_ORIGIN_WHITELIST = (
    'http://localhost:8081',
)

# Application definition

INSTALLED_APPS = [
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    'rest_framework',
    'corsheaders',
    'app',
    'cliente.apps.ClienteConfig',
    'instalacion.apps.InstalacionConfig',
    'camara_zona.apps.CamaraZonaConfig',
    'monitor.apps.MonitorConfig',
]

MIDDLEWARE = [
    'django.middleware.security.SecurityMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.locale.LocaleMiddleware',
    'django.middleware.common.CommonMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
    'whitenoise.middleware.WhiteNoiseMiddleware',
]

ROOT_URLCONF = 'core.urls'
LOGIN_REDIRECT_URL = "home"   # Route defined in app/urls.py
LOGOUT_REDIRECT_URL = "home"  # Route defined in app/urls.py
TEMPLATE_DIR = os.path.join(BASE_DIR, "core/templates")  # ROOT dir for templates

TEMPLATES = [
    {
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'DIRS': [TEMPLATE_DIR],
        'APP_DIRS': True,
        'OPTIONS': {
            'context_processors': [
                'django.template.context_processors.debug',
                'django.template.context_processors.request',
                'django.contrib.auth.context_processors.auth',
                'django.contrib.messages.context_processors.messages',
                'instalacion.context_processors.side_menu_instalaciones',
            ],
        },
    },
]



WSGI_APPLICATION = 'core.wsgi.application'

# Database
# https://docs.djangoproject.com/en/3.0/ref/settings/#databases

DATABASES = {
         'default': {
             'ENGINE': 'djongo',
             'NAME': 'SMF_BDControlAforo',
             'ENFORCE_SCHEMA': False,
             'CLIENT': {
                 'host': 'mongodb://smfactory:S0ftwar4Mf4ac0ory@5.196.27.225:27117',
                 'port': 27117,
                 'username': 'smfactory',
                 'password': 'S0ftwar4Mf4ac0ory',
                 'authSource': 'admin',
                 'authMechanism': 'SCRAM-SHA-1'
             }
        },
        'BDMosayk': {
             'ENGINE': 'djongo',
             'NAME': 'SMF_BDMosayk',
             'ENFORCE_SCHEMA': False,
             'CLIENT': {
                 'host': 'mongodb://smfactory:S0ftwar4Mf4ac0ory@127.0.0.1:27117',
                 'port': 27117,
                 'username': 'smfactory',
                 'password': 'S0ftwar4Mf4ac0ory',
                 'authSource': 'admin',
                 'authMechanism': 'SCRAM-SHA-1'
             }
        },
        'ControlAforo': {
            'ENGINE': 'djongo',
            'NAME': 'SMF_BDControlAforoMQTTMensajes',
            'CLIENT': {
                 'host': 'mongodb://smfactory:S0ftwar4Mf4ac0ory@5.196.27.225:27117',
                 'port': 27117,
                 'username': 'smfactory',
                 'password': 'S0ftwar4Mf4ac0ory',
                 'authSource': 'admin',
                 'authMechanism': 'SCRAM-SHA-1',
                 'db_coleccion': 'camara_mvsense_messages'
            }
        },
}


# Password validation
# https://docs.djangoproject.com/en/3.0/ref/settings/#auth-password-validators

AUTH_PASSWORD_VALIDATORS = [
    {
        'NAME': 'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.MinimumLengthValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.CommonPasswordValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.NumericPasswordValidator',
    },
]


# Internationalization
# https://docs.djangoproject.com/en/3.0/topics/i18n/

LANGUAGE_CODE = 'en-us'

TIME_ZONE = 'UTC'

USE_I18N = True

USE_L10N = True

USE_TZ = True

#############################################################
# SRC: https://devcenter.heroku.com/articles/django-assets

BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))

# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/1.9/howto/static-files/
STATIC_ROOT = os.path.join(BASE_DIR, 'staticfiles')
STATIC_URL = '/static/'

MEDIA_ROOT = os.path.join(BASE_DIR, 'media')
MEDIA_URL = '/media/'


# Extra places for collectstatic to find static files.
STATICFILES_DIRS = (
    os.path.join(BASE_DIR, 'core/static'),
)


#Allow Log by email or console
ADMINS = (('Ligia', 'ligiapuertas@gmail.com'), ('Joel', 'joelsegoviacrespo@gmail.com'))
LOGGING = {
    'version': 1,
    'disable_existing_loggers': True,
    'formatters': {
        'verbose': {
            'format': '%(asctime)s %(levelname)s [%(name)s:%(lineno)s] %(module)s %(process)d %(thread)d %(message)s'
        }
    },
    'handlers': {
        'gunicorn': {
            'level': 'DEBUG',
            'class': 'logging.handlers.RotatingFileHandler',
            'formatter': 'verbose',
            'filename': '/opt/gunicorn.errors',
            'maxBytes': 1024 * 1024 * 100,  # 100 mb
        }
    },
    'loggers': {
        'gunicorn.errors': {
            'level': 'DEBUG',
            'handlers': ['gunicorn'],
            'propagate': True,
        },
    }
}

#EMAIL_BACKEND = "django.core.mail.backends.filebased.EmailBackend"
#EMAIL_FILE_PATH = os.path.join(BASE_DIR, "sent_emails")

EMAIL_BACKEND = 'django.core.mail.backends.console.EmailBackend'

#EMAIL_HOST = 'smtp.sendgrid.net'
#EMAIL_PORT = 587
#EMAIL_HOST_USER = 'testsite_app'
#EMAIL_HOST_PASSWORD = 'mys3cr3tp4ssw0rd'
#EMAIL_USE_TLS = True
#DEFAULT_FROM_EMAIL = 'TestSite Team <noreply@example.com>'
